encrypted-media/scripts/keystatuses.js

```diff
@@ -0,0 +1,161 @@
+function runTest(config)
+{
+    var testname = config.keysystem + ', keystatuses';
+
+    var configuration = getSimpleConfigurationForContent( config.content );
+
+    if ( config.initDataType && config.initData ) configuration.initDataTypes = [ config.initDataType ];
+
+    async_test(function(test)
+    {
+        var mediaKeySession;
+        var initDataType;
+        var initData;
+        var closed = false;
+
+        // Even though key ids are uint8, using printable values so that
+        // they can be verified easily.
+        var key1 = new Uint8Array( config.content.keys[ 0 ].kid ),
+            key2 = new Uint8Array( config.content.keys[ 1 ].kid ),
+            key1String = arrayBufferAsString(key1),
+            key2String = arrayBufferAsString(key2);
+
+        function onFailure(error) {
+            forceTestFailureFromPromise(test, error);
+        }
+
+        function processMessage(event)
+        {
+            // No keys added yet.
+            assert_equals(mediaKeySession.keyStatuses.size, 0);
+
+            waitForEventAndRunStep('keystatuseschange', mediaKeySession, processKeyStatusesChange, test);
+
+            // Add keys to session
+            config.messagehandler( event.messageType, event.message ).then( function( response ) {
+                event.target.update( response ).catch(onFailure);
+            });
+        }
+
+        function checkKeyStatusFor2Keys()
+        {
+            // Two keys added, so both should show up in |keyStatuses|.
+            assert_equals(mediaKeySession.keyStatuses.size, 2);
+
+            // Check |keyStatuses| for 2 entries.
+            var result = [];
+            for (let item of mediaKeySession.keyStatuses) {
+                result.push({ key: arrayBufferAsString(item[0]), value: item[1] });
+            }
+            function lexicographical( a, b ) { return a < b ? -1 : a === b ? 0 : +1; }
+            function lexicographicalkey( a, b ) { return lexicographical( a.key, b.key ); }
+            var expected = [{ key: key1String, value: 'usable'}, { key: key2String, value: 'usable'}].sort( lexicographicalkey );
+            assert_object_equals(result,
+                                 expected,
+                                 'keyStatuses fails');
```

Comment:
Can the assert text here be made more explicit?  Instead of 'keystatuses fails', could it say 'keystatuses should return usable' or something like that?

I will mark other examples of this below.